test-schema.json

```diff
@@ -1,26 +1,25 @@
 {
-    "$schema": "http://json-schema.org/draft-06/schema#",
-    "description": "Schema for tests",
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "description": "A schema for files contained within this suite",
+
     "type": "array",
+    "minItems": 1,
     "items": {
+        "description": "An individual test case, containing multiple tests of a single schema's behavior",
+
         "type": "object",
         "required": [ "description", "schema", "tests" ],
         "properties": {
-            "id": {
-                "description": "Uniquely identifies a set of tests",
-                "type": "string",
-                "format": "uri"
-            },
             "description": {
-                "description": "The test set description",
+                "description": "The test case description",
                 "type": "string"
             },
             "comment": {
-                "description": "Any additional comments about the test set",
+                "description": "Any additional comments about the test case",
                 "type": "string"
             },
             "schema": {
-                "description": "This should be a valid schema. This should be a ref to a meta-schema if schema keywords need testing."
+                "description": "A valid JSON Schema (one written for the corresponding version directory that the file sits within)."
```

Comment:
If we add an enum property to the test files, we could probably key off of that and enforce a specific metaschema here.

Reply:
As in provide some other mechanism for indicating a metaschema test? We could, but each of these changes is delicate, since someone's existing test runner still may not be expecting that. Obviously we're going to need to do something like breaking our informal backwards compatibility anyhow to resolve the `optional/` folder situation, so we could in theory do so for that too, but lemme know if I misunderstood the idea entirely.

Reply:
No, I mean actually validate the schemas in the test files.  If the test file specifies what version schemas its for (because JSON Schema does see the file path) then we can use that to select a metaschema to validate the test schema against.

I was thinking something like the below, but then I realized the test file is an array, and you'd have to specify the version on every test...  Might be more headache than it's worth now that I see it.

---

Leaving for posterity

In the test file
```
{
  "version": "2020-12",
   ... // the rest of the test file
}
```

In the above schema:
```
{
  ...
  "oneOf": [
    {
      "properties": {
        "version": { "const": "2020-12" },
        "schema": { "$ref": "<2020-12 metaschema>" }
    },
    ... // more for other versions
  ]
  ...
}
```